### PR TITLE
add hook, change hook location and change private $error to public $error

### PR DESCRIPTION
**Pull request by handoyo**
I change the private $error to public $error in the checkout/address as i found out that i cannot add additional error message from our extension.

As for the checkout/confirm.tpl, i moved the hook due to i put the hook in the wrong location.

### CODE REVIEW

File: public_html/storefront/view/default/template/embed/checkout/confirm.tpl [hunk 1]
```diff
@@ -81,8 +82,9 @@
 			<?php if ($payment_method) { ?>
 				<tr>
 					<td class="align_left"><?php echo $payment_firstname . ' ' . $payment_lastname; ?>
-						<br/><?php echo $telephone; ?></td>
-			<?php echo $this->getHookVar('checkout_confirm_payment_info_1'); ?>
+						<br/><?php echo $telephone; ?>
+						<?php echo $this->getHookVar('checkout_confirm_payment_info_1'); ?>
```
Comment (abolabo) [inline public_html/storefront/view/default/template/embed/checkout/confirm.tpl:86]: We pasted hookvar here for adding new column. 
In your variant you changing column content. 
Why do you do that way? Why not via updateData()  (i mean changing variable $telephone inside aview) ?

File: public_html/extensions/fast_checkout/storefront/view/default/template/responses/checkout/payment.tpl [hunk 2]
```diff
@@ -180,7 +180,7 @@ $guest_data = $this->session->data['fc']['guest'];
                                                 <td style="vertical-align: middle;">
                                                     <label for="<?php
                                                             $idd = str_replace('.', '', $quote['id']);
-                                                                echo preg_replace('/[^a-zA-Z0-9\.-_]/', '', $idd.$quote['id']); ?>"
+                                                                echo preg_replace('/[^a-zA-Z0-9\.-_]/', '', $idd.$quote['id']); ?>_title"
```
Comment (abolabo) [inline public_html/extensions/fast_checkout/storefront/view/default/template/responses/checkout/payment.tpl:183]: please clarify what is for

**Review by abolabo — CHANGES_REQUESTED** [review]

File: public_html/extensions/fast_checkout/storefront/view/default/template/responses/checkout/payment.tpl (hunk 2 above)
Comment (handoyo) [reply public_html/extensions/fast_checkout/storefront/view/default/template/responses/checkout/payment.tpl:183]: One of our shipper extension changed the shipping method title via ajax. The extra "_title" was used so i can target the correct label via javascript.

File: public_html/storefront/view/default/template/embed/checkout/confirm.tpl (hunk 1 above)
Comment (handoyo) [reply public_html/storefront/view/default/template/embed/checkout/confirm.tpl:86]: We need to add extra content under the telephone and within the same td wrapper instead of outside the telephone wrapper.

Comment (abolabo) [reply public_html/storefront/view/default/template/embed/checkout/confirm.tpl:86]: You don't need to move hookvar inside TD tag for that. 
Why do you do not changing $telephone variable via $that->view->assign('telephone', $telephone + $your_extra_html_data ); ? You can append some data to telephone via UpdatData method.
i'm afraid your changes int this place affects on other ready extensions that uses this hookVar too.

Comment (handoyo) [reply public_html/storefront/view/default/template/embed/checkout/confirm.tpl:86]: I will try your suggestion first and see the result.

File: public_html/extensions/fast_checkout/storefront/view/default/template/responses/checkout/payment.tpl (hunk 2 above)
Comment (abolabo) [reply public_html/extensions/fast_checkout/storefront/view/default/template/responses/checkout/payment.tpl:183]: i think this changes broke labeling. You changed attribute "for" of LABEL tag but not changed attribute "id" of INPUT tag (see code below). When you click on shipping icon it WILL NOT select radiobutton. Can you check? Just set some icon for shipping extension to see

Comment (handoyo) [reply public_html/extensions/fast_checkout/storefront/view/default/template/responses/checkout/payment.tpl:183]: Yes i can check it. i will do that along with testing the update data you suggested just now.

Comment (handoyo) [reply public_html/extensions/fast_checkout/storefront/view/default/template/responses/checkout/payment.tpl:183]: I tried adding a shipping icon but the icon was not shown in the page.

File: public_html/storefront/view/default/template/embed/checkout/confirm.tpl (hunk 1 above)
Comment (handoyo) [reply public_html/storefront/view/default/template/embed/checkout/confirm.tpl:85]: I have followed your suggestion and it works. And i have changed the code.
